src/app/components/recommendations/recommendatons.directive.spec.js

```diff
@@ -0,0 +1,80 @@
+(function() {
+  'use strict';
+  describe('kil-recommendations', function() {
+    var directiveElem, scope, state, userService, sectionsService;
+
+    beforeEach( function(){
+      module( 'healthGuide' );
+
+      inject( function ($compile, $rootScope, _userService_, _sectionsService_) {
+        scope = $rootScope.$new();
+        directiveElem = $compile('<kil-recommendations/>')(scope);
+        userService = _userService_;
+        sectionsService = _sectionsService_;
+        state = 'nutrition';
+        scope.layout = {
+          user: userService.user,
+          data: sectionsService.data[state]
+        };
+        scope.$digest();
+      });
+    });
+
+    it('should applied template', function () {
+      expect(directiveElem.html()).not.toEqual('');
+    });
+
+
+    it('should compile text', function () {
+      expect(directiveElem.find('h2').text()).toContain('Health recommendations for ');
```

Comment:
you should try to be independent from the markup and the hardcoded text,
all you want to know is:
* if the directive is rendered or not.
* if the directive displays all the values
* if the directive displays correct values for every case (ng-show, ng-if)

btw, try to use ```ng-if```, it's better for performance reasons